lib/array_intersection.rb

```diff
@@ -1,3 +1,3 @@
 def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
+  intersect = list1 & list2
```

Comment:
That works as it uses a feature of Ruby.  Do think about how you'd solve it with a hash.